streams/src/main/java/org/apache/kafka/streams/state/internals/CachingWindowStore.java

```diff
@@ -216,5 +217,58 @@ private V fetchPrevious(final Bytes key, final long timestamp) {
             }
         }
     }
+    
+    @Override
+    public KeyValueIterator<Windowed<Bytes>, byte[]> all() {
+        validateStoreOpen();
 
+        final KeyValueIterator<Windowed<Bytes>, byte[]>  underlyingIterator = underlying.all();
+        final ThreadCache.MemoryLRUCacheBytesIterator cacheIterator = cache.all(name);
+
+        return new MergedSortedCacheWindowStoreKeyValueIterator(
+            cacheIterator,
+            underlyingIterator,
+            bytesSerdes,
+            windowSize,
+            cacheFunction
+        );
+    }
+    
+    @Override
+    public KeyValueIterator<Windowed<Bytes>, byte[]> fetchAll(long timeFrom, long timeTo) {
+        validateStoreOpen();
+        
+        final KeyValueIterator<Windowed<Bytes>, byte[]> underlyingIterator = underlying.fetchAll(timeFrom, timeTo);
+        final ThreadCache.MemoryLRUCacheBytesIterator cacheIterator = cache.all(name);
+        
+        final NavigableSet<Bytes> allBytes = cache.keySet(name);
+        if (allBytes == null) return null;
+        
+        Bytes keyFrom = new Bytes(new byte[0]);
+        Bytes keyTo = null;
+        for (final Bytes bytes : allBytes) {
+            final long timestamp = WindowStoreUtils.timestampFromBinaryKey(bytes.get());
+            final Bytes key = WindowStoreUtils.bytesKeyFromBinaryKey(SegmentedCacheFunction.bytesFromCacheKey(bytes));
+            
+            if (keyFrom == new Bytes(new byte[0]) && timestamp >= timeFrom) {
```

Comment:
Can't we remove this? i.e, can `keyFrom` just be `new Bytes(new byte[0])`?  I don't think it needs to change as we already have the starting timestamp.